bin/citrea/tests/bitcoin_e2e/tests/prover_test.rs

```diff
@@ -98,7 +106,163 @@ impl TestCase for BasicProverTest {
+            node_url: format!(
+                "http://127.0.0.1:{}/wallet/{}",
+                da_config.rpc_port,
+                NodeKind::Bitcoin
+            ),
+            node_username: da_config.rpc_user.clone(),
+            node_password: da_config.rpc_password.clone(),
+            network: bitcoin::Network::Regtest,
+            da_private_key: Some(secret_key),
+        };
+        let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
+        let bitcoin_da_service = Arc::new(
+            BitcoinService::new_with_wallet_check(
+                bitcoin_da_service_config,
+                RollupParams {
+                    reveal_light_client_prefix: REVEAL_LIGHT_CLIENT_PREFIX.to_vec(),
+                    reveal_batch_prover_prefix: REVEAL_BATCH_PROOF_PREFIX.to_vec(),
+                },
+                tx,
+            )
+            .await
+            .unwrap(),
+        );
+        bitcoin_da_service.clone().spawn_da_queue(rx);
+
+        // Generate confirmed UTXOs
+        da.generate(10, None).await?;
+
+        let seq_height0 = sequencer.client.eth_block_number().await;
+        assert_eq!(seq_height0, 0);
+
+        for _ in 0..10 {
+            sequencer.client.send_publish_batch_request().await;
+        }
+
+        da.generate(5, None).await?;
+
+        // Wait for blob inscribe tx to be in mempool
+        da.wait_mempool_len(1, None).await?;
+
+        da.generate(5, None).await?;
+        let finalized_height = da.get_finalized_height().await?;
+        prover
+            .wait_for_l1_height(finalized_height, Some(Duration::from_secs(300)))
+            .await;
+
+        da.generate(5, None).await?;
+        let proofs = full_node
+            .wait_for_zkproofs(finalized_height + 5, Some(Duration::from_secs(120)))
+            .await
+            .unwrap();
+
+        assert!(proofs
+            .first()
+            .unwrap()
+            .state_transition
+            .preproven_commitments
+            .is_empty());
+
+        // Fetch the commitment created from the previous L1 range
+        let commitments: Vec<SequencerCommitment> = sequencer
+            .client
+            .ledger_get_sequencer_commitments_on_slot_by_number(15)
+            .await
+            .unwrap()
+            .unwrap()
+            .into_iter()
+            .map(|response| SequencerCommitment {
+                merkle_root: response.merkle_root,
+                l2_start_block_number: response.l2_start_block_number,
+                l2_end_block_number: response.l2_end_block_number,
+            })
+            .collect();
+
+        let fee_sat_per_vbyte = bitcoin_da_service.get_fee_rate().await.unwrap();
+        bitcoin_da_service
+            .send_transaction_with_fee_rate(
+                None,
+                DaData::SequencerCommitment(commitments.first().unwrap().clone()),
+                fee_sat_per_vbyte,
+            )
+            .await
+            .unwrap();
+
```

Comment:
right here, also trigger a sequencer commitment
and wait for mempool len 4

so that the prover will generate a proof for that block leaving out preproven commmitment (the one you sent)